src/components/bar/bar.js

```diff
@@ -995,6 +992,74 @@ Bar.prototype = {
     this.setSelected(options, true);
   },
 
+  /**
+   * Action to happen on click.
+   * @private
+   * @param {object} d - The data object
+   * @param {number} i - The index
+   * @param {object} selector - The selector element
+   * @returns {void}
+   */
+  doClickAction(d, i, selector) {
+    const self = this;
+    const s = this.settings;
+    const isSelected = selector && d3.select(selector).classed('is-selected');
+    const thisGroupId = parseInt(d3.select(selector.parentNode).attr('data-group-id'), 10);
+
+    charts.setSelectedElement({
+      task: (isSelected ? 'unselected' : 'selected'),
+      container: self.element,
+      selector,
+      isTrigger: self.initialSelectCall ? false : !isSelected,
+      triggerGroup: s.isGrouped,
+      d,
+      i,
+      type: s.type,
+      dataset: s.dataset,
+      isSingle: self.isSingular,
+      isGrouped: s.isGrouped,
+      isStacked: s.isStacked,
+      svg: self.svg,
+      clickedLegend: s.clickedLegend
+    });
+
+    if (isSelected && !self.initialSelectCall) {
+      self.element.triggerHandler('selected', [d3.select(selector).nodes(), {}, (s.isGrouped ? thisGroupId : i)]);
+    }
+  },
+
+  /**
+   * Action to happen on double click.
+   * @private
+   * @param {object} d - The data object
+   * @param {number} i - The index
+   * @param {object} selector - The selector element
+   * @returns {void}
+   */
+  doDoubleClickAction(d, i, selector) {
+    const self = this;
+    const s = this.settings;
+    let args;
+    if (s.isGrouped) {
+      const groupElem = d3.select(selector.parentNode);
+      const groupIndex = parseInt(groupElem?.attr('data-group-id'), 10);
+      const groupItems = [];
+      groupElem.selectAll('.bar').each(function (d2, i2) {
+        groupItems.push({ elem: this, data: d2, index: i2 });
+      });
+      args = [{ groupElem: groupElem.node(), groupIndex, groupItems }];
+    } else if (s.type === 'bar-stacked') {
+      args = [];
+      self.svg.selectAll('.series-group').each(function () {
+        const bar = d3.select(this).selectAll('.bar').nodes()[i];
+        args.push({ elem: bar, data: d3.select(bar).datum(), index: i });
+      });
+    } else {
+      args = [{ data: d, index: i, elem: selector }];
+    }
+    self.element.triggerHandler('eventdblclick', [args]);
```

Comment:
```suggestion
    self.element.triggerHandler('eventdblclick', [args]);
```

Ok see - yes lets make this dblclick so it matches the names in other components

Reply:
updated now, to be `dblclick`